### ISSUE TITLE
vcpkg does not work properly due to lib files not copied to the installed folder

### ISSUE THREAD

**Issue by matusnovak** [mosra/magnum-plugins#80]
I am having problem installing magnum plugins via the latest vcpkg command line tool. The library is built correctly but the libraries (for example `FreeTypeFont.lib`) are not copied to the target directory where all of the other vcpkg installed libraries are located in `C:\tools\vcpkg\installed\x64-windows\lib\magnum\fonts`. The only file located in the installed directory is `lib\magnum\fonts\MagnumFont.lib`.

But, I was able to find the missing lib files in the build folder `C:\tools\vcpkg\buildtrees\magnum-plugins\x64-windows-rel\Release\lib\magnum\fonts`

Due to this problem (for example using a freetype plugin) the command `find_package(MagnumPlugins REQUIRED FreeTypeFont)` will always fail with an error `Could NOT find MagnumPlugins (missing: FreeTypeFont)`. The actual command that fails is `find_library` from here: https://github.com/mosra/magnum-plugins/blob/master/modules/FindMagnumPlugins.cmake#L241-L246 because those lib files are missing.

Once I copy the lib files from the `buildtrees` into the `installed` folder, the CMake completes without an issue.

Additionally, after the build only few DLL files were copied to the output directory. `importers/*.dll` and `imageconverters/*.dll` DLL files were copied over but `fonts/*.dll` were not at all. Had to manually copy them over.

TL;DR: After installing magnum-plugins via vcpkg the *.lib files are in the buildtrees but not copied to the installed directory.

**Comment by mosra** [comment]
Hi!

In case you are building dynamic libs (which is the default), this is the intended behavior -- only `*.dll`s are installed, no `*.lib`s, as the plugins are meant to be used through the `PluginManager` (and not explicitly linked to). Explicit linking is meant to be used in case the plugins are built as static, and in that case the needed `*.lib` files get properly installed.

There was some additional vcpkg-specific issue why the `*.lib` files had to be deleted for dynamic plugins and only the `*.dll`s kept, but since those weren't meant to be linked to anyway (except in very rare cases of inter-plugin dependencies), that really wasn't an issue in practice.

> but `fonts/*.dll` were not at all

Hmm, but this doesn't look right. Are you installing the package with fonts enabled (`vcpkg install magnum-plugins[core,freetypefont,...]`)? Those are not enabled by default, so the package doesn't build/install them if you do just `vcpkg install magnum-plugins`. Looking at [the portfile](https://github.com/microsoft/vcpkg/blob/ef60f1d9b97fab6fcc61651936196866bb6553d0/ports/magnum-plugins/portfile.cmake#L73-L87), there isn't any special handling for importers / fonts, so this seems to be the reason.

**Comment by matusnovak** [comment]
Hi @mosra 

Ah, thank you. I did not realise those are meant to be loaded at runtime if compiled dynamically through the PluginManager. I have switched to static build of plugins and that solved all of my problems.

And yes I was installing it via `vcpkg install magnum-plugins[core,freetypefont,...]` but I was trying to explicitly link them, therefore expecting CMake to copy them to the project output folder, my bad. Was doing that at midnight and probably should read the documentation more.

Switched to `vcpkg install magnum-plugins[...]:x64-windows-static` and solved my issue.

Thanks a lot for a quick reply!
